R/select_helpers.R

```diff
@@ -32,38 +28,46 @@
   }
 
   # check if pattern is a function like "starts_with()"
-  select <- tryCatch(eval(p), error = function(e) NULL)
-  exclude <- tryCatch(eval(p2), error = function(e) NULL)
+  if (!is.null(p) && !insight::safe_deparse(p) %in% ls(globalenv())) {
+    select <- tryCatch(eval(p), error = function(e) NULL)
+  } else {
+    select <- NULL
+  }
+  if (!is.null(p2) && !insight::safe_deparse(p2) %in% ls(globalenv())) {
```

Comment:
Same as above.

It would actually be better to create a Boolean variable to track this instead of such complex conditional expression.

Reply:
I can define a boolean before `if` but not sure it's easier to understand. I went with `p_is_expr_not_in_globenv` but not the best name ever